lib/Tumblr/StreamBuilder/StreamRankers/WeightedRandomStreamRanker.php

```diff
@@ -51,8 +51,12 @@ protected function rank_inner(array $stream_elements, StreamTracer $tracer = nul
         foreach ($valid_elements as $element) {
             /** @var RecommendationLeafStreamElementTrait $original_element */
             $original_element = $element->get_original_element();
-            // calculate sampling score
-            $r = pow(mt_rand() / $max_rand, (1 / $original_element->get_score()));
+            if ($original_element->get_score() === 0.0) {
```

Comment:
just as a note, if for any reason this comes as 0 instead, 0===0.0 evaluates to false